### ISSUE TITLE
Access env step info in callbacks

### ISSUE THREAD

**Issue by ManifoldFR** [DLR-RM/stable-baselines3#137]
**Describe the bug**

I would like to define a callback where I can access the `info` returned by `env.step()`. In my use case, it's because I'd like to log reward sub-terms.

**Code example**

I have my `env.step()` setup as
```python

def step(self):
    ...
    total_reward = root_weight * root_reward + \
                   pose_weight * pose_reward + \
                   vel_weight * vel_reward
    info = {}
    info['reward'] = dict(
        root_reward=root_reward,
        pose_reward=pose_reward,
        vel_reward=vel_reward)
    ...
    return obs, reward, done, info

```
I know you can access the field `self.model.ep_info_buffer` in every class inheriting from `BaseCallback`, but it doesn't seem to be populated with the data I need, because the `_update_info_buffer()` method of `BaseAlgorithm` only keeps the `episode` key.

**Comment by PartiallyTyped** [comment]
Related to #115.
I need to add a couple of tests and merge to master but info will be available in callbacks through `self.locals_['infos']`.

**Comment by ManifoldFR** [comment]
This would be perfect, thanks!

**Comment by araffin** [comment]
If you need to log things from the env, best is to use a env wrapper, in fact the Monitor wrapper can log most things.

**Comment by ManifoldFR** [comment]
The Monitor wrapper seems to record things at the end of every episode, which is not really what I need if I want to aggregate the data and send it to Tensorboard.  
Also, shouldn't `make_vec_env` have an option to pass arguments to the Monitor?

**Comment by araffin** [comment]
> The Monitor wrapper seems to record things at the end of every episode, which is not really what I need if I want to aggregate the data and send it to Tensorboard

True, and if you need tensorboard then go for a custom callback.

> Also, shouldn't `make_vec_env` have an option to pass arguments to the Monitor?

This is in the doc ;) You would need to pass `wrapper_kwargs` (if I recall) in addition to `Monitor` as the wrapper class.

**Comment by ManifoldFR** [comment]
I just saw commit 9003a09d5bb48c8b9e94d862d7303f62764427a7 come in. Thanks a lot!

**Comment by Miffyli** [comment]
Closing this issue as resolved via #115.
